packages/code/src/Code.tsx

```diff
@@ -169,16 +290,60 @@ function Code({
     </Syntax>
   );
 
+  function handleScroll(e: React.UIEvent) {
+    const { scrollWidth, clientWidth: elementWidth } = e.currentTarget;
+    const isScrollable = scrollWidth > elementWidth;
+
+    if (isScrollable) {
+      const scrollPosition = e.currentTarget.scrollLeft;
+      const maxPosition = scrollWidth - elementWidth;
+
+      if (scrollPosition > 0 && scrollPosition < maxPosition) {
+        setScrollState(ScrollState.Both);
+      } else if (scrollPosition > 0) {
+        setScrollState(ScrollState.Left);
+      } else if (scrollPosition < maxPosition) {
+        setScrollState(ScrollState.Right);
+      }
+    }
+  }
+
   if (!multiline) {
     return (
       <div className={wrapperStyle}>
         {renderedWindowChrome}
 
         <div
-          {...(rest as DetailedElementProps<HTMLDivElement>)}
-          className={wrapperClassName}
+          className={css`
+            display: flex;
+          `}
         >
-          {renderedSyntaxComponent}
+          <div
+            {...(rest as DetailedElementProps<HTMLDivElement>)}
+            className={wrapperClassName}
+            onScroll={handleScroll}
```

Comment:
Do we want to [debounce this? ](https://davidwalsh.name/javascript-debounce-function) Generally good practice for onScroll events.